### ISSUE TITLE
Do not assign null to listitems when iterating over a list

### ISSUE THREAD

**Issue by RalfKoban** [RalfKoban/MiKo-Analyzers#307]
It may happen that developers want to clear a list and its contents. In such cases you can see code like following:

```C#
var list = new List<object>();

foreach (var item in list)
{
    item = null; // this is already caught by CS1656 (https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-messages/cs1656)
}

list.ForEach(item => item = null);

list.ForEach(item => { item = null; });
```

That code does not do what the developer tries to acchieve.
The developer tries to set `null` inside the list, so that the list will only contain a number of `null` values.

However, as `item` is always a copy, only the copy (reference) will be set to `null` and not the original value.

So we should warn about it.
